src/three/plugins/SUBTREELoader.js

```diff
@@ -0,0 +1,747 @@
+			 The last three elements (indices 9, 10, and 11) define the z axis direction and half-length.
+			 */
+
+
+			boundingVolume = [...this.rootTile.boundingVolume.box];
+			const scale = Math.pow(2, -(tile.__depth ?? tile.__level));
+
+
+			const vecCenter = new Vector3();
+			vecCenter.fromArray(boundingVolume.slice(0, 3));
+
+			const modelSpaceX = -1 + (2 * tile.__x + 1) * scale;
+			const modelSpaceY = -1 + (2 * tile.__y + 1) * scale;
+			let modelSpaceZ = 0;
+
+			let rootHalfAxes = new Matrix3();
+			rootHalfAxes.fromArray(boundingVolume, 3);
+			const scaleFactors = new Vector3(scale, scale, 1);
+
+			if (!isNaN(tile.__z)) {
+				modelSpaceZ = -1 + (2 * tile.__z + 1) * scale;
+				scaleFactors.z = scale;
+			}
+
+
+			let center = new Vector3(
+				modelSpaceX,
+				modelSpaceY,
+				modelSpaceZ
+			);
+			center = center.applyMatrix3(rootHalfAxes);
+			center = center.add(vecCenter);
+
+
+			//todo implement better multiply
+			rootHalfAxes = rootHalfAxes.toArray();
+			rootHalfAxes[0] *= scaleFactors.x;
+			rootHalfAxes[1] *= scaleFactors.x;
+			rootHalfAxes[2] *= scaleFactors.x;
+			rootHalfAxes[3] *= scaleFactors.y;
+			rootHalfAxes[4] *= scaleFactors.y;
+			rootHalfAxes[5] *= scaleFactors.y;
+			rootHalfAxes[6] *= scaleFactors.z;
+			rootHalfAxes[7] *= scaleFactors.z;
+			rootHalfAxes[8] *= scaleFactors.z;
+
+			let res = [];
+			center = center.toArray();
+			res = res.concat(center)
+			res = res.concat(rootHalfAxes)
+
+			return {box: res}
+
+		}
+
+
+	}
+
+	/**
+	 * Each child’s geometricError is half of its parent’s geometricError
+	 * @param {Object | SubtreeTile} tile
+	 * @return {number}
+	 */
+	getGeometricError(tile) {
+		return this.rootTile.geometricError / Math.pow(2, tile.__depth ?? tile.__level);
+	}
+
+
+	/**
+	 * Determine what child subtrees exist and return a list of information
+	 *
+	 * @param {Object} subtree The subtree for looking up availability
+	 * @param {Array} bottomRow The bottom row of tiles in a transcoded subtree
+	 * @returns {[]} A list of identifiers for the child subtrees.
+	 * @private
+	 */
+	listChildSubtrees(subtree, bottomRow) {
+		const results = [];
+		const branchingFactor = this.rootTile.__subtreeDivider;
+		for (let i = 0; i < bottomRow.length; i++) {
+			const leafTile = bottomRow[i];
+			if (leafTile === undefined) {
+				continue;
+			}
+
+			for (let j = 0; j < branchingFactor; j++) {
+
+				const index = i * branchingFactor + j;
+				if (this.getBit(subtree._childSubtreeAvailability, index)) {
+					results.push({
+						tile: leafTile,
+						childMortonIndex: index
+					});
+				}
+			}
+		}
+		return results;
+	}
+
+
+	parseImplicitURI(tile, basePath, uri) {
+		uri = uri.replace("{level}", (tile.__depth ?? tile.__level) ?? 0);
+		uri = uri.replace("{x}", tile.__x ?? 0);
+		uri = uri.replace("{y}", tile.__y ?? 0);
+		uri = uri.replace("{z}", tile.__z ?? 0);
+		uri = new URL(uri, basePath + '/').toString();
+		return uri;
+	}
```

Comment:
We shouldn't need to pass the base path or anything here. Lets just replace the "level", "x", "y", and "z" values. Then the url will be expanded to a full uri in the `preprocessNode` function.